apps/web/src/pages/settings/organization/OrganizationLogo.tsx

```diff
@@ -0,0 +1,80 @@
+import { errorMessage, successMessage } from '@novu/design-system';
+import { IResponseError } from '@novu/shared';
+import { useMutation } from '@tanstack/react-query';
+import axios from 'axios';
+import React from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { useUpdateOrganizationBranding } from '../../../api/hooks';
+import { getSignedUrl } from '../../../api/storage';
+import { ProfileImage } from '../../../components/shared';
+
+type FormValues = {
+  logo: string;
+};
+
+const MIME_TYPES = {
+  'image/jpeg': 'jpeg',
+  'image/png': 'png',
+};
+
+export function OrganizationLogo({ logo }: { logo?: string }) {
+  const { updateOrganizationBranding } = useUpdateOrganizationBranding();
+  const { control, reset } = useForm<FormValues>({
+    defaultValues: {
+      logo: logo,
+    },
+  });
+
+  const { mutateAsync: getSignedUrlAction } = useMutation<
+    { signedUrl: string; path: string; additionalHeaders: object },
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Should `additionalHeaders` be required?‏

Reply:
Yes, it adds some Cloud Provider specific headers specially in the self-hosted env where users are flexible to replace S3 with Azure blob or GCP's storage.